python/tests/res/enkf/jobs/test_enkf_scaling_job.py

```diff
@@ -0,0 +1,647 @@
+    assert enkf_scaling_pca._get_nr_primary_components(primary_components, 0.99) == 6
+
+
+def test_filter_on_column_index():
+    matrix = np.random.rand(10, 10)
+
+    df = pd.DataFrame(matrix)
+
+    assert df.shape == (10, 10)
+
+    index_lists = [[0, 1], [1, 2, 3], [1, 2, 3, 4, 5]]
+    for index_list in index_lists:
+        result_df = enkf_scaling_get_data._filter_on_column_index(df, index_list)
+        assert result_df.shape == (10, len(index_list))
+
+    with pytest.raises(IndexError):
+        enkf_scaling_get_data._filter_on_column_index(df, [11])
+
+
+
+@pytest.mark.equinor_test
+def test_main_entry_point_summary_data_calc(tmpdir):
+    arguments = {
+        "CALCULATE_KEYS": {"keys": ["WWCT:OP_1","WWCT:OP_2"]},
+    }
+    cwd = os.getcwd()
+    test_data_dir = os.path.join(_TEST_DATA_DIR, "Equinor","config","obs_testing")
+    tmpdir.chdir()
+    import subprocess
+    subprocess.call(["cp", "-r", test_data_dir, "."])
+    res_config = ResConfig("obs_testing/config")
+    ert = EnKFMain(res_config)
+    obs = ert.getObservations()
+
+    obs_vector = obs["WWCT:OP_2"]
+
+    for index, node in enumerate(obs_vector):
+            assert node.getStdScaling(index) == 1.0
+
+    enkf_scaling_job.main_entry_point(ert, arguments)
+
+    for index, node in enumerate(obs_vector):
+            assert node.getStdScaling(index) == np.sqrt(61.0*2.0)
+
+    arguments["CALCULATE_KEYS"].update({"index":[1, 2, 3]})
+
+    enkf_scaling_job.main_entry_point(ert, arguments)
+
+    for index, node in enumerate(obs_vector):
+        if index in arguments["CALCULATE_KEYS"]["index"]:
+            assert node.getStdScaling(index) == np.sqrt(3.0*2.0)
+        else:
+            assert node.getStdScaling(index) == np.sqrt(61.0 * 2.0)
+
+    os.chdir(cwd)
+
+@pytest.mark.equinor_test
+def test_main_entry_point_summary_data_update(tmpdir):
+    arguments = {
+        "CALCULATE_KEYS": {"keys": ["WWCT:OP_1","WWCT:OP_2"]},
+        "UPDATE_KEYS": {"keys": [{"key": "WWCT:OP_2", "index": [1, 2, 3, 4, 5]}]},
+    }
+
+    cwd = os.getcwd()
+    test_data_dir = os.path.join(_TEST_DATA_DIR, "Equinor","config","obs_testing")
+    tmpdir.chdir()
+    import subprocess
+    subprocess.call(["cp", "-r", test_data_dir, "."])
+    res_config = ResConfig("obs_testing/config")
+    ert = EnKFMain(res_config)
+
+    obs = ert.getObservations()
+    obs_vector = obs["WWCT:OP_2"]
+
+    enkf_scaling_job.main_entry_point(ert, arguments)
+
+    for index, node in enumerate(obs_vector):
+        if index in arguments["UPDATE_KEYS"]["keys"][0]["index"]:
+            assert node.getStdScaling(index) == np.sqrt(61.0*2.0)
+        else:
+            assert node.getStdScaling(index) == 1.0
+
+    obs_vector = obs["WWCT:OP_1"]
+
+    enkf_scaling_job.main_entry_point(ert, arguments)
+
+    for index, node in enumerate(obs_vector):
+        assert node.getStdScaling(index) == 1.0
+
+    os.chdir(cwd)
+
+
+@pytest.mark.equinor_test
```

Comment:
Could me regular test, if we use the snake oil data set as a REFCASE

Reply:
:+1: